state/sqlserver/sqlserver.go

```diff
@@ -440,23 +440,23 @@ func (s *SQLServer) Delete(req *state.DeleteRequest) error {
 		res, err = s.db.Exec(s.deleteWithoutETagCommand, sql.Named(keyColumnName, req.Key))
 	}
 
+	// err represents errors thrown by the stored procedure or the database itself
 	if err != nil {
-		if req.ETag != nil {
-			return state.NewETagError(state.ETagMismatch, err)
-		}
-
 		return err
 	}
 
+	// if the key is not found, then the stored procedure returns 0 rows affected
 	rows, err := res.RowsAffected()
 	if err != nil {
 		return err
 	}
 
-	if rows != 1 {
-		return fmt.Errorf("items was not updated")
+	// When an ETAG is specified, a row must have been deleted or else we return an ETag mismatch error
+	if req.ETag != nil && rows != 1 {
```

Comment:
Should this also treat empty string req.ETag as nil ETag for consistency with executeSet?
```suggestion 
	if rows != 1 && req.ETag != nil && *req.ETag != "" {
```

Reply:
Yes probably.